### ISSUE TITLE
More clarity on default units

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#4738]
As I dug a bit deeper yesterday to finish up #4734 I re-learned a few things.  In the ancient days we had a way to set US versus SI as a global parameter.  I believe this lived in a gmt.conf separate file (back when we used .gmtdefaults4 for settings).  Users could then switch from SI to US if they so desired, and could decide which they wanted.  The GMT4 installer supported **--enable-US** to do this.
As we have sought to pull GMT out of its inch/US-centric upbringing into true scientific SI units we no longer have that option.  What remains is a Cmake parameter in the _ConfigUserAdvancedTemplate.cmake_ file that is commented out by default:

`#set (UNITS "US")`

with the default being SI.  So 100% of any installers built by us and (most likley 100%) of any user-installation built via CMake will have SI as the default system.  That **UNITS** setting ends up defining **GMT_DEF_UNITS** during the build and it is then either "SI" or "US".  If "US" then this is the only line in GMT source code that checks it:

```
if (!strncmp (GMT_DEF_UNITS, "US", 2U))
	gmtinit_conf_classic_US (GMT);	/* Override with US settings */
```

This exchanges happens at the end of setting the initial GMT defaults (in _gmtinit_conf_classic_).  So I think it is correct to say that GMT has SI units as default (since that is how the build process works) and hence cm is the default unit for all dimensions except fonts and pen thickness where points are the default unit.  So **-Sc**1 means a 1cm-diameter circle, **-W**1 means **-W**1p, and 12,Helvetica means 12p,Helvetica.  Users can always add their preferred units to override this, e.g., **-Sc**0.25i, **-W**0.2c and 1c,Times-Roman.  And, of course, users can change **PROJ_LENGTH_UNITS** to whatever but that does not change the meaning of what default is - that is an override.

I will have a look at the **UNITS** section in the general features in the Cookbook to make sure it is in alignment with this reality.

**Comment by meghanrjones** [comment]
The cookbook text seems clear to me and I do not think any revisions are needed. Here's the relevant section regarding PROJ_LENGTH_UNIT and defaults:

> GMT programs accept plot dimensional quantities (widths, offsets, etc.) in cm, inch, or point (1/72 of an inch). There are two ways to ensure that GMT understands which unit you intend to use:
>     1. Append the desired unit to the dimension you supply. This way is explicit and clearly communicates what you intend, e.g., -JM10c means the map width being passed to the -JM switch is 10 cm, and modifier +o24p means we are offsetting a feature by 24 points from its initial location.
>     2. Set the parameter PROJ_LENGTH_UNIT to the desired unit. Then, all dimensions without explicit units will be interpreted accordingly. By default, GMT always initializes PROJ_LENGTH_UNIT to cm.
> The latter method is less robust as other users may have a different default unit set and then your script may not work as intended. For portability, we therefore recommend you always append the desired unit explicitly.

**Comment by PaulWessel** [comment]
OK thanks, I agree.  Closing this issue.

**Comment by seisman** [comment]
> cm is the default unit for all dimensions except fonts and pen thickness where points are the default unit.

I'm afraid we never mention that the default units of fonts and pen thickness are points.

**Comment by meghanrjones** [comment]
> I'm afraid we never mention that the default units of fonts and pen thickness are points.

You're right. I can create a PR with that specified.

**Comment by seisman** [comment]
Closed by #4739
